File: backend/src/main/java/com/project/entities/VideoSession.java
package com.project.entities;

import java.time.Instant;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

import com.project.enums.VideoSessionStatus;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(
    name = "video_sessions"
)
@Data
public class VideoSession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long videoSessionId;

    @OneToOne
    @JoinColumn(name = "appointment_id", nullable = false)
    private Appointment appointment;

    @Column(nullable = false, unique = true)
    private String roomToken;

    private LocalDateTime allowedFrom;
    private LocalDateTime allowedUntil;

    private LocalDateTime patientJoined;
    private LocalDateTime professionalJoined;

    @CreationTimestamp
    private Instant createdAt;

    @Enumerated(EnumType.STRING)
    private VideoSessionStatus status;
}

File: backend_dotnet/SafeMind.Core/Entities/VideoSession.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using SafeMind.Core.Enums;

namespace SafeMind.Core.Entities
{
    [Table("video_sessions")]
    public class VideoSession
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("video_session_id")]
        public long VideoSessionId { get; set; }

        [Required]
        [ForeignKey(nameof(Appointment))]
        [Column("appointment_id")]
        public long AppointmentId { get; set; }

        public Appointment Appointment { get; set; } = null!;

        [Required]
        [Column("room_token")]
        public string RoomToken { get; set; } = string.Empty;

        [Column("allowed_from")]
        public DateTime? AllowedFrom { get; set; }
        [Column("allowed_until")]
        public DateTime? AllowedUntil { get; set; }

        [Column("patient_joined")]
        public DateTime? PatientJoined { get; set; }
        [Column("professional_joined")]
        public DateTime? ProfessionalJoined { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        [Column("status")]
        public VideoSessionStatus Status { get; set; }
    }
}
